### DISCUSSION TITLE
Rows in banded/shaded table being skipped

### DISCUSSION THREAD

**Discussion by ashleych** [jsvine/pdfplumber#548] [Ask for help with specific PDFs]
I have a shaded table, with every alternate row shaded. Only the shaded rows seem to get picked. 
The image below is the original image
![orignal](https://user-images.githubusercontent.com/3990319/143869099-1e7369b7-5e70-46c0-afcc-8065645ae3b6.png)

Code being attempted is

```
ts = {
    "vertical_strategy": "lines",
    "horizontal_strategy": "lines"}

p0.extract_tables(table_settings=ts)
```

Output obtained is : (every second row is skipped). Any pointers would be very helpful

![failedExtraction](https://user-images.githubusercontent.com/3990319/143869262-f140639a-12ce-4c1b-82cb-2031e0046d70.png)

**Comment by samkit-jain** [comment]
Hi @ashleych Appreciate your interest in the library. For such cases, you can have a high value for `join_tolerance` (start with a high value like 50 and then either go down or up depending on the table). That will connect the vertical lines and fill out the gap. But, it will miss out on the last row. For that, instead of `lines` as the horizontal strategy, use `text`.

**Reply by ashleych** [reply]
```
ts = {
    "vertical_strategy": "lines",
    "horizontal_strategy": "text",
    "join_tolerance":50
    }
```
THank you so much for your quick response. The above indeed helped, but last line is still not detected. Also multi line cells are now getting split into multiple rows,which isnt desired
![failedExtraction_1](https://user-images.githubusercontent.com/3990319/143892373-561c5673-7610-42c1-ac5b-a332ba0d1c68.png)

**Reply by samkit-jain** [reply]
Ah! You are facing the same issue as #265 I can think of 4 workarounds for it as of now
1. You fix the core issue and open a PR.
2. Downgrade pdfplumber version to 0.5.22.
3. Crop the page to remove the bottom "The accompanying ..." text and then run table extraction on it.
4. Use `explicit` horizontal strategy with `explicit_horizontal_lines` being
    ```python
    all_horizontal_lines = []
    
    for word in page.extract_words(extra_attrs=["top", "bottom"]):
        all_horizontal_lines.append(word["top"])
        all_horizontal_lines.append(word["bottom"])

    all_horizontal_lines = sorted(set(all_horizontal_lines))
    ```
    Above, what I am essentially doing is getting all the top and bottom Y coordinates of all the words to essentially reproduce the same (well, almost) logic as in v0.5.22. The table settings will then become
    ```python
    ts = {
        "vertical_strategy": "lines",
        "horizontal_strategy": "explicit",
        "explicit_horizontal_lines": all_horizontal_lines,
        "join_tolerance": 50
    }
    ```
    You might have to tweak it a bit depending on your needs.

**Reply by ashleych** [reply]
@samkit-jain thanks once again, will check out your options. Just curious, #256 leads to a link that didnt seem to describe the issue.

**Reply by samkit-jain** [reply]
My bad. Updated the reply. Meant #265 and not #256

**Reply by adicognext** [reply]
Hi, @samkit-jain I tried 3 workarounds out of the 4 you have mentioned, still no luck with the results. Can you please help me out here?
## 1) Cropping the image.
![Screenshot from 2022-01-11 09-49-18](https://user-images.githubusercontent.com/90697470/148881806-ffaf0304-e790-4c85-a5c5-85030c7f0e21.png)
## 2) Downgrading to version 0.5.22.
![Screenshot from 2022-01-11 10-03-19](https://user-images.githubusercontent.com/90697470/148881955-bf1847a0-2406-43e1-8f3a-ab8951218d7c.png)
## 3) Trying out the settings you've mentioned.
![Screenshot from 2022-01-11 09-50-10](https://user-images.githubusercontent.com/90697470/148882022-a4aaa908-dece-4982-89f7-379ea1fa8fa0.png)


In the first 2 cases, I have used the settings as:
ts = {
    "vertical_strategy": "lines",
    "horizontal_strategy": "text",
    "join_tolerance":50
    }

also, I tried the same on a different pdf with the same structure and the problem persists.
Thank you so much in advance.

**Reply by samkit-jain** [reply]
Hi @adicognext Could you please share the PDF you are using and the page number redacting any sensitive information from the PDF so that I can assist better. Please also include your expected output. Looking at the screenshots, what happens when you try using `snap_tolerance` (to merge those multiple close lines into one) and `intersection_tolerance` (to pick up the last row).

**Reply by adicognext** [reply]
### The concerned pdf.
[002_3D Systems Corporation.pdf](https://github.com/jsvine/pdfplumber/files/7879038/002_3D.Systems.Corporation.pdf)
### The expected output is where the last row is also getting recognized.

**Reply by samkit-jain** [reply]
Thanks for sharing the PDF. I was able to get the desired results by using the following extraction settings
```python
{
    "vertical_strategy": "lines",
    "horizontal_strategy": "text",
    "snap_tolerance": 5,
    "intersection_tolerance": 15,
}
```
Result when running on page 6 with pdfplumber 0.5.22
![image](https://user-images.githubusercontent.com/15127115/149769820-a07dffa8-67a5-4217-b322-662b88df3fe6.png)

**Reply by adicognext** [reply]
Hii, Thank you so much for your help, this is exactly what was expected.
